### ISSUE TITLE
Difference in Results between local and hosted machine

### ISSUE THREAD

**Issue by yxoos** [pytorch/pytorch#54832]
Hello,

I am training an algorithm on my machine and also using Google Colab's GPU to train on bigger models. However, the accuracies I get on my machine are by a magnitude different, I thought this problem was due to the version difference and realized it is not after I updated my machine's torch to 1.8.0. I am out of ideas. I seed everything before I instantiate the model as well. Any ideas?

Thank you,

**Comment by imaginary-person** [comment]
I'm not a PyTorch team member, so here's an unofficial answer.
You didn't specify any details about your code or your PyTorch environment, so I'm just speculating about various factors that could've resulted in the disparity.

Google Colab uses GPUs of multiple Nvidia architectures - Turing (T4), Kepler (Tesla K80), and Pascal (P4, P100).
Which GPU had you used on Colab? You can't select a particular GPU model to use but you can view the model name.
Some PyTorch code differs for various GPU architectures, so you should compare your GPU's architecture with Colab's.
Otherwise, it'd be hard to figure out if there's some bug in your code (or maybe PyTorch), if results would be different (in this case, for example).

Please note that PyTorch 1.8.0 on Google Colab has been custom-built with CUDA Toolkit 10.1, although it's recommended that PyTorch 1.8.0 be installed with either CUDA toolkit 10.2 or 11.1, so your PyTorch installation probably uses a more recent CUDA Toolkit, which could have resulted in a difference, especially if some bug was fixed in CUDA toolkit 10.2 that affects your code.

**Comment by H-Huang** [comment]
Thanks for raising this. I agree with the above poster that this could be caused by a variety of issues. It would be helpful if you could provide the code of the model you are running as well as run this script (https://github.com/pytorch/pytorch/blob/master/torch/utils/collect_env.py) in both environments so that we can see what the difference is. Please reraise the issue if you can get this information.
